### PR TITLE
[megatron] fix: optimizer scheduler misalignment with FSDP

### PR DESCRIPTION
**Pull request by ETOgaosion**
### What does this PR do?

Fix learning rate divergence with FSDP, megatron.training's default lr decay policy is linear, but FSDP has not supported this, so return back to `constant`.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### High-Level Design

> Demonstrate the high-level design if this PR is complex.

### Specific Changes

> List the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl?tab=readme-ov-file#contribution-guide).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl?tab=readme-ov-file#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ).

### CODE REVIEW

File: verl/utils/logger/aggregate_logger.py [hunk 1]
```diff
@@ -25,9 +25,13 @@
 
 def concat_dict_to_str(dict: Dict, step):
     output = [f"step:{step}"]
+    extended_length_keys = {"actor/lr": 7}
```
Comment (vermouth1992) [inline verl/utils/logger/aggregate_logger.py]: This is hacky. We need a systematic way to handle this

Comment (vermouth1992) [reply verl/utils/logger/aggregate_logger.py]: Maybe give an optional field of precision for each metric?

Comment (ETOgaosion) [reply verl/utils/logger/aggregate_logger.py]: Can we use `pprint` instead? Seems use scientific notation is OK.

```
>>> import pprint
>>> pprint.pp(0.000000001)
1e-09
```

Comment (ETOgaosion) [reply verl/utils/logger/aggregate_logger.py]: I wonder if this is acceptable, more precise but a little untidy:

```
step:1 - global_seqlen/min:52887 - global_seqlen/max:59028 - global_seqlen/minmax_diff:6141 - global_seqlen/balanced_min:56446 - global_seqlen/balanced_max:56447 - global_seqlen/mean:56446.125 - actor/entropy:1.9626522064208984 - critic/grad_norm:941.3387909444281 - critic/lr:2e-05 - critic/vf_loss:26.17556804418564 - critic/vf_clipfrac:0.32582305930554867 - critic/vpred_mean:1.4724534023553133 - perf/mfu/critic:0.2226467589063418 - actor/kl_loss:0.0004533928334958067 - actor/kl_coef:0.0010000000000000002 - actor/pg_loss:0.00028897453050472235 - actor/pg_clipfrac:0.001040518200208776 - actor/ppo_kl:-5.7683862623710076e-05 - actor/pg_clipfrac_lower:0.0 - actor/grad_norm:0.9848380312787193 - perf/mfu/actor:0.12533451802376736 - perf/max_memory_allocated_gb:4.433852195739746 - perf/max_memory_reserved_gb:5.0234375 - perf/cpu_memory_used_gb:0.00018310546875 - actor/lr:1e-06 - training/global_step:1 - training/epoch:0 - critic/score/mean:3.6824328899383545 - critic/score/max:10.75 - critic/score/min:-13.5 - critic/rewards/mean:3.6824328899383545 - critic/rewards/max:10.75 - critic/rewards/min:-13.5 - critic/advantages/mean:-6.883408332214458e-08 - critic/advantages/max:4.276647567749023 - critic/advantages/min:-6.650888442993164 - critic/returns/mean:3.327622652053833 - critic/returns/max:10.75 - critic/returns/min:-13.5 - critic/values/mean:-1.712753415107727 - critic/values/max:16.25 - critic/values/min:-19.625 - critic/vf_explained_var:-1.138066291809082 - response_length/mean:346.2587890625 - response_length/max:512.0 - response_length/min:3.0 - response_length/clip_ratio:0.390625 - prompt_length/mean:94.7265625 - prompt_length/max:179.0 - prompt_length/min:56.0 - prompt_length/clip_ratio:0.0 - timing_s/generate_sequences:24.798080444335938 - timing_s/reshard:2.8393430709838867 - timing_s/gen:27.933526224980596 - timing_s/reward:5.981584483990446 - timing_s/old_log_prob:13.337011502997484 - timing_s/ref:9.156857737019891 - timing_s/values:4.93116571099381 - timing_s/adv:0.03139720298349857 - timing_s/update_critic:8.573562754987506 - timing_s/update_actor:15.000672547990689 - timing_s/save_checkpoint:13.02682404301595 - timing_s/step:98.06855953900958 - timing_per_token_ms/values:0.010920071375567875 - timing_per_token_ms/gen:0.07878163693097985 - timing_per_token_ms/update_actor:0.03321900428946781 - timing_per_token_ms/update_critic:0.018986163255200215 - timing_per_token_ms/ref:0.02027787057353337 - timing_per_token_ms/adv:6.952913726030478e-05 - perf/total_num_tokens:451569 - perf/time_per_step:98.06855953900958 - perf/throughput:575.578200244156
```

**Review by vermouth1992 — APPROVED** [review]
